sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -22,14 +22,67 @@
 
 from __future__ import absolute_import
 
-from apache_beam import coders
+from apache_beam.io.iobase import Read
+from apache_beam.io.iobase import Write
 from apache_beam.runners.dataflow.native_io import iobase as dataflow_io
+from apache_beam.transforms import PTransform
+from apache_beam.transforms import ParDo
 from apache_beam.transforms.display import DisplayDataItem
 
-__all__ = ['PubSubSink', 'PubSubSource']
 
+__all__ = ['ReadStringsFromPubSub', 'WriteStringsToPubSub']
 
-class PubSubSource(dataflow_io.NativeSource):
+
+class ReadStringsFromPubSub(PTransform):
```

Comment:
That was my initial approach by I wasn't able to figure out how to set the coder on the intermediate PCollection. Finally I didn't think using a coder is what we wanted because we are trying to stop using coders for parsing input and formatting output in IOs since that is the biggest reason why nested/outer context encoding exists.

Also, this mirrors the Java approach of having an explicit readStrings/readAvro/readMessage. I could see later that we would expose ReadAvroFromPubSub and ReadMessagesFromPubSub in the future.